### PR TITLE
Render noData if TreeView has no matches after filtering (T562910)

### PR DESCRIPTION
**Pull request by sometimesElen**

### CODE REVIEW

File: js/ui/tree_view/ui.tree_view.search.js [hunk 1]
```diff
@@ -53,11 +53,14 @@ var TreeViewSearch = TreeViewBase.inherit(extend({}, searchBoxMixin, { _dataSour
     },
 
     _repaintContainer: function() {
-        var $container = this.$element().find("." + NODE_CONTAINER_CLASS).first();
+        var $container = this.$element().find("." + NODE_CONTAINER_CLASS).first(),
+            rootNodes;
 
         if($container) {
```
Comment (DokaRus) [inline js/ui/tree_view/ui.tree_view.search.js]: Always true. Use `$container.length` instead

File: js/ui/collection/ui.collection_widget.base.js [hunk 2]
```diff
@@ -995,10 +995,12 @@ var CollectionWidget = Widget.inherit({
...
     },
 
-    _renderEmptyMessage: function() {
+    _renderEmptyMessage: function(items) {
         var noDataText = this.option("noDataText"),
-            items = this.option("items"),
-            hideNoData = !noDataText || (items && items.length) || this._isDataSourceLoading();
+            hideNoData;
+
+        items = items || this.option("items"),
```
Comment (DokaRus) [inline js/ui/collection/ui.collection_widget.base.js]: Why comma?

File: js/ui/collection/ui.collection_widget.base.js [hunk 3]
```diff
@@ -995,10 +995,12 @@ var CollectionWidget = Widget.inherit({
         return this._itemContainer();
     },
 
-    _renderEmptyMessage: function() {
+    _renderEmptyMessage: function(items) {
         var noDataText = this.option("noDataText"),
```
Comment (DokaRus) [inline js/ui/collection/ui.collection_widget.base.js]: Better to move `items` to top

```
items = items || this.option("items");
var noDataText = this.option("noDataText"),
    hideNoData = !noDataText || (items && items.length) || this._isDataSourceLoading();

**Review by DokaRus — CHANGES_REQUESTED** [review]

File: js/ui/collection/ui.collection_widget.base.js (hunk 2 above)
Comment (sometimesElen) [reply js/ui/collection/ui.collection_widget.base.js]: Oops...

File: js/ui/collection/ui.collection_widget.base.js (hunk 3 above)
Comment (sometimesElen) [reply js/ui/collection/ui.collection_widget.base.js]: Fixed

File: js/ui/tree_view/ui.tree_view.search.js (hunk 1 above)
Comment (sometimesElen) [reply js/ui/tree_view/ui.tree_view.search.js]: Fixed

**Review by DokaRus — APPROVED** [review]

**Pull request merged**